core/src/main/java/com/linecorp/armeria/server/AnnotatedHttpServiceConfigurator.java

```diff
@@ -0,0 +1,164 @@
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+
+import com.linecorp.armeria.internal.annotation.AnnotatedHttpService;
+import com.linecorp.armeria.server.annotation.ExceptionHandlerFunction;
+import com.linecorp.armeria.server.annotation.RequestConverterFunction;
+import com.linecorp.armeria.server.annotation.ResponseConverterFunction;
+
+/**
+ * A configurator to configure {@link ExceptionHandlerFunction}s, {@link RequestConverterFunction}s
+ * or {@link ResponseConverterFunction}s for an {@link AnnotatedHttpService}.
+ */
+final class AnnotatedHttpServiceConfigurator {
+
+    /**
+     * Creates a new instance with the specified {@code exceptionHandlersAndConverters}.
+     */
+    static AnnotatedHttpServiceConfigurator ofExceptionHandlersAndConverters(
+            Iterable<?> exceptionHandlersAndConverters) {
+
+        Builder<ExceptionHandlerFunction> exceptionHandlers = null;
+        Builder<RequestConverterFunction> requestConverters = null;
+        Builder<ResponseConverterFunction> responseConverters = null;
+
+        for (final Object o : exceptionHandlersAndConverters) {
+            if (o instanceof ExceptionHandlerFunction) {
+                if (exceptionHandlers == null) {
+                    exceptionHandlers = ImmutableList.builder();
+                }
+                exceptionHandlers.add((ExceptionHandlerFunction) o);
+            } else if (o instanceof RequestConverterFunction) {
+                if (requestConverters == null) {
+                    requestConverters = ImmutableList.builder();
+                }
+                requestConverters.add((RequestConverterFunction) o);
+            } else if (o instanceof ResponseConverterFunction) {
+                if (responseConverters == null) {
+                    responseConverters = ImmutableList.builder();
+                }
+                responseConverters.add((ResponseConverterFunction) o);
+            } else {
+                throw new IllegalArgumentException(o.getClass().getName() +
+                                                   " is neither an exception handler nor a converter.");
+            }
+        }
+
+        final List<ExceptionHandlerFunction> exceptionHandlerFunctions =
+                exceptionHandlers != null ? exceptionHandlers.build() : ImmutableList.of();
```

Comment:
then we can just do `exceptionHandlers.build()` because when there's no element in it, it calls `ImmutableList.of()`.